torchvision/datasets/fgvc_aircraft.py

```diff
@@ -0,0 +1,106 @@
+            and returns a transformed version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+    """
+
+    _URL = "https://www.robots.ox.ac.uk/~vgg/data/fgvc-aircraft/archives/"
+    _URL_FILE = "fgvc-aircraft-2013b.tar.gz"
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "trainval",
+        download: bool = False,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+    ) -> None:
+        super().__init__(root, transform=transform, target_transform=target_transform)
+        self._split = verify_str_arg(split, "split", ("train", "val", "trainval", "test"))
+
+        self._data_path = os.path.join(self.root, "fgvc-aircraft-2013b")
+        if download:
+            self._download()
+
+        if not self._check_exists():
+            raise RuntimeError("Dataset not found. You can use download=True to download it")
+
+        self._label_names = self._get_label_names(self._data_path)
+
+        # Parse the downloaded files
+        self._image_folder = os.path.join(self.root, self._split)
+
+        self._image_files = []
+        self._labels = []
+        self._label_name_to_idx = dict(zip(self._label_names, range(len(self._label_names))))
+
+        self._read_fgvc_aircrafts_images_labels(self._data_path)
+
+    def __len__(self) -> int:
+        return len(self._image_files)
+
+    def __getitem__(self, idx) -> Tuple[Any, Any]:
+        image_file, label = self._image_files[idx], self._labels[idx]
+        image = PIL.Image.open(image_file).convert("RGB")
+
+        if self.transform:
+            image = self.transform(image)
+
+        if self.target_transform:
+            label = self.target_transform(label)
+
+        return image, label
+
+    def _download(self):
+        """
+        Download the FGVC Aircraft dataset archive and extract it under root.
+        """
+        if self._check_exists():
+            return
+        download_and_extract_archive(self._URL + self._URL_FILE, self.root)
+
+    def _check_exists(self) -> bool:
+        return os.path.exists(self._data_path) and os.path.isdir(self._data_path)
+
+    def _read_fgvc_aircrafts_images_labels(self, input_path: str):
+        image_data_folder = os.path.join(input_path, "data", "images")
+        labels_path = os.path.join(input_path, "data", f"images_variant_{self._split}.txt")
+
+        with open(labels_path, "r") as labels_file:
+            lines = [line.strip() for line in labels_file]
+            for line in lines:
+                line_list = line.split(" ")
+                image_name = line_list[0]
+                label_name = self._parse_aircraft_name(" ".join(line_list[1:]))
+                self._labels.append(self._label_name_to_idx[label_name])
+                self._image_files.append(os.path.join(image_data_folder, image_name + str(".jpg")))
```

Comment:
Given that this method again does nothing complicated, is quite short, and is only used once, I think we can inline it in `__init__`. This will also resolve the linting errors the CI is currently complaining about.